go/sqltypes/result.go

```diff
@@ -324,16 +330,14 @@ func (result *Result) StripMetadata(incl querypb.ExecuteOptions_IncludedFields)
 // to another result.Note currently it doesn't handle cases like
 // if two results have different fields.We will enhance this function.
 func (result *Result) AppendResult(src *Result) {
-	if src.RowsAffected == 0 && len(src.Rows) == 0 && len(src.Fields) == 0 {
-		return
+	result.RowsAffected += src.RowsAffected
+	if src.InsertIDUpdated() {
+		result.InsertID = src.InsertID
 	}
+	result.InsertIDChanged = result.InsertIDUpdated() || src.InsertIDUpdated()
```

Comment:
nit: `result` is accumulating here, it does not need to check `InsertIDUpdated` with a function call `result.InsertIDUpdated()`